File: vendror_android/app/src/main/java/magentoegypt/locafy/base_app/base/RetrofitInstance.java
package magentoegypt.locafy.base_app.base;

import androidx.annotation.NonNull;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.io.IOException;
import java.util.Objects;

import okhttp3.Credentials;
import okhttp3.Dispatcher;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {

    public static Retrofit instance;

    public static Retrofit getInstance() {
        if (null == instance)
            createNewInstance();
        return instance;
    }

    private static void createNewInstance() {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client;
       // String credential = Credentials.basic("locafy", "Market2025");
        client = new OkHttpClient.Builder().addInterceptor(new Interceptor() {
            @NonNull
            @Override
            public okhttp3.Response intercept(Interceptor.Chain chain) throws IOException {

                Request originalRequest = chain.request();
                Request requestWithUserAgent = originalRequest.newBuilder()
                      //  .header("Authorization", credential)
                        .addHeader("Cache-Control","no-cache")
                        .addHeader("User-Agent", Objects.requireNonNull(System.getProperty("http.agent")))
                        .build();
                return  chain.proceed(requestWithUserAgent);


            }
        }).build();
     //   OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
        Dispatcher dispatcher = new Dispatcher();
        dispatcher.setMaxRequests(1);
     //   httpClient.addInterceptor(logging);
     //   httpClient.dispatcher(dispatcher);

        Gson gson = new GsonBuilder()
                .setLenient()
                .create();

        instance = new Retrofit.Builder()
                .baseUrl(AppUrl.getBaseUrl())
                .addConverterFactory(GsonConverterFactory.create(gson))
                .client(client)
                .build();


    }
}

File: vendror_android/app/src/main/java/magentoegypt/locafy/addons/advance_Report/appBase/RetrofitInstance.kt
package magentoegypt.locafy.addons.advance_Report.appBase

import com.google.gson.GsonBuilder
import okhttp3.Credentials
import okhttp3.Dispatcher
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object RetrofitInstance {
    var baseUrl: String? = null
    private var instance: Retrofit? = null
    fun getInstance(url: String?): Retrofit? {
        baseUrl = url
        if (null == instance) createNewInstance()
        return instance
    }

    private fun createNewInstance() {
        val loggingInterceptor = HttpLoggingInterceptor()
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
        val basicAuthInterceptor = Interceptor { chain ->
            val originalRequest: Request = chain.request()
          //  val credential = Credentials.basic("locafy", "Market2025")  // Automatically Base64 encodes

            val newRequest = originalRequest.newBuilder()
             //   .header("Authorization", credential)
                .header("Content-Type", "application/json")
                .build()

            chain.proceed(newRequest)
        }
        val httpClient = OkHttpClient.Builder()
        httpClient.connectTimeout(100, TimeUnit.SECONDS)
            .writeTimeout(100, TimeUnit.SECONDS)
            .readTimeout(300, TimeUnit.SECONDS)
            .addInterceptor(loggingInterceptor)
            .addInterceptor(basicAuthInterceptor)

        val dispatcher = Dispatcher()
        dispatcher.maxRequests = 1
        val gson = GsonBuilder()
            .setLenient()
            .create()
        instance = Retrofit.Builder()
            .baseUrl(baseUrl!!)
            .addConverterFactory(GsonConverterFactory.create(gson))
            .client(httpClient.build())
            .build()
    }
}
